app/services/boarding_service.rb

```diff
@@ -74,9 +74,13 @@ def ensure_values
       spaceship = Spaceship::Tunes.login(@user, @password)
```

Comment:
I think you could raise the error before this line if user, password, or app_id are empty right?